src/main/java/seedu/address/model/item/IdContainsNumberPredicate.java

```diff
@@ -0,0 +1,32 @@
+package seedu.address.model.item;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Item}'s {@code Id} matches any of the keywords given.
+ * Only output items with id that matches exactly with the query
+ */
+public class IdContainsNumberPredicate implements Predicate<Item> {
```

Comment:
The class seems like a repeat of `NameContainsKeywordPredicate`, maybe we can implement a generic class that can handle both `name` and `id`? 

e.g. an `ItemFilter` class that can be configured to filter by name or by class

Reply:
Also, the current name suggests that Ids contain numbers. Which might not be the case?